tests/testthat/test-derive_param_extreme_record.R

```diff
@@ -0,0 +1,153 @@
+        new_vars = exprs(
+          ADT = convert_dtc_to_dt(CMSTDTC),
+          AVALC = CMDECOD
+        )
+      ),
+      records_source(
+        dataset_name = "pr",
+        filter = PRDECOD == "ACS",
+        new_vars = exprs(
+          ADT = convert_dtc_to_dt(PRSTDTC),
+          AVALC = PRDECOD
+        )
+      )
+    ),
+    source_datasets = list(cm = cm, pr = pr),
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(ADT),
+    mode = "first",
+    set_values_to = exprs(
+      PARAMCD = "FIRSTACT",
+      PARAM = "First Anti-Cancer Therapy"
+    )
+  )
+
+  expect_dfs_equal(expected_output, actual_output, keys = c("USUBJID", "PARAMCD", "PARAM", "ADT", "AVALC")) # nolint
+})
+
+## Test 2: Error order variable not inside source datasets ----
+test_that("derive_param_extreme_record Test 2: Error order variable not inside source datasets", {
+  aevent <- tibble::tribble(
+    ~STUDYID, ~USUBJID,     ~LBSTDTC, ~PARAMCD, ~PARAM,
+    "1001",        "1", "2023-01-01",    "TST", "TEST",
+    "1001",        "2", "2023-01-01",    "TST", "TEST",
+    "1001",        "3", "2023-01-01",    "TST", "TEST"
+  )
+
+  cm <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~CMDECOD,     ~CMSTDTC,
+    "1001",        "1",    "ACT", "2020-12-25"
+  )
+
+  pr <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~PRDECOD,     ~PRSTDTC,
+    "1001",        "1",    "ACS", "2020-12-27",
+    "1001",        "2",    "ACS", "2021-12-25",
+    "1001",        "3",    "ACS", "2022-12-25",
+  )
+  expect_error(
+    derive_param_extreme_record(
+      dataset = aevent,
+      sources = list(
+        records_source(
+          dataset_name = "cm",
+          filter = CMDECOD == "ACT",
+          new_vars = exprs(
+            ADT = convert_dtc_to_dt(CMSTDTC),
+            AVALC = CMDECOD
+          )
+        ),
+        records_source(
+          dataset_name = "pr",
+          filter = PRDECOD == "ACS",
+          new_vars = exprs(
+            ADT = convert_dtc_to_dt(PRSTDTC),
+            AVALC = PRDECOD
+          )
+        )
+      ),
+      source_datasets = list(cm = cm, pr = pr),
+      by_vars = exprs(STUDYID, USUBJID),
+      order = exprs(ADT2),
+      mode = "first",
+      set_values_to = exprs(
+        PARAMCD = "FIRSTACT",
+        PARAM = "First Anti-Cancer Therapy"
+      )
+    )
+  )
+})
+
+## Test 3: Sources is not in proper list format ----
+test_that("derive_param_extreme_record Test 3: Sources is not in proper list format", {
+  aevent <- tibble::tribble(
+    ~STUDYID, ~USUBJID,     ~LBSTDTC, ~PARAMCD, ~PARAM,
+    "1001",        "1", "2023-01-01",    "TST", "TEST",
+    "1001",        "2", "2023-01-01",    "TST", "TEST",
+    "1001",        "3", "2023-01-01",    "TST", "TEST"
+  )
+
+  pr <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~PRDECOD,     ~PRSTDTC,
+    "1001",        "1",    "ACS", "2020-12-27",
+    "1001",        "2",    "ACS", "2021-12-25",
+    "1001",        "3",    "ACS", "2022-12-25",
+  )
+  expect_error(
```

Comment:
Please add `regexp` or `class` argument to verify the the expected error is issued.